docs/advanced/patterns_by_usecase.md

```diff
@@ -709,23 +709,38 @@ const UsageComponent = () => (
 
 [View in the TypeScript Playground](https://www.typescriptlang.org/play/?jsx=2#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wFgAoCmATzCTgAUcwBnARjgF44BvOTCBABccFjCjAAdgHM4AXwDcVWvSYRWAJi684AIxRQRYiTPlLK5TAFdJGYBElwAstQDCuSJKSSYACjDMLCJqrBwAPoyBGgCUvBRwcMCYcL4ARAIQqYmOAeossTzxCXAA9CVwuawAdPpQpeVIUDhQRQlEMFZQjgA8ACbAAG4AfDyVLFUZct0l-cPmCXJwSAA2LPSF5MX1FYETgtuNza1w7Z09syNjNQZTM4ND8-IUchRoDmJwAKosKNJI7uAHN4YCJkOgYFUAGKubS+WKcIYpIp9e7HbouAGeYH8QScdKCLIlIZojEeIE+PQGPG1QnEzbFHglABUcHRbjJXgpGTxGSytWpBlSRO2UgGKGWwF6cCZJRe9OmFwo0QUQA)
 
-Further reading: [how to ban passing `{}` if you have a `NoFields` type.](http://www.javiercasas.com/articles/typescript-impossible-states-irrepresentable)
-
-## Props: Must Pass Both
+## Props: Pass nothing or all
 
 ```tsx
-type OneOrAnother<T1, T2> =
-  | (T1 & { [K in keyof T2]?: undefined })
-  | (T2 & { [K in keyof T1]?: undefined });
+interface Nothing {
+  [key: string]: never; // needed because an empty interface is equivalent to `{}`
```

Comment:
We should link to a section that explains why `{}` is bad. It's not intuitive why `{}` is bad.

Also: should this be `[key: keyof any]`? 

Would also be nice to check what others recommend instead of `{}`. I've seen `Record<keyof any, never>` as well so if that's more popular, I'd stick with that to avoid fragmentation.

Reply:
I dug a little deeper and this is a bit of a rabbit hole. The most commonly used way seems to be `Record<string, never>`, and that's also what [typescript-eslint recommends](https://typescript-eslint.io/rules/ban-types/) as default:

> If you want a type meaning "empty object", you probably want `Record<string, never>` instead.

But it does not seem like this is recommended by the TypeScript team, since `never` semantically means something else: https://github.com/microsoft/TypeScript/issues/47486#issuecomment-1015671856

I added dedicated sections about these object types to make it easier to link to them. I also added alternative approaches that might be better to avoid having to rely on typing an empty object.

What do you think?